web/api/v1/api.go

```diff
@@ -1740,6 +1742,40 @@ func TSDBStatsFromIndexStats(stats []index.Stat) []TSDBStat {
 	return result
 }
 
+func (api *API) serveTSDBBlock(_ *http.Request) apiFuncResult {
+	dirs, err := os.ReadDir(api.dbDir)
+	if err != nil {
+		return apiFuncResult{nil, &apiError{errorInternal, fmt.Errorf("read TSDB dir: %w", err)}, nil, nil}
+	}
+
+	var blocks []tsdb.BlockMeta
+
+	for _, entry := range dirs {
+		if !entry.IsDir() {
+			continue
+		}
+		_, err := ulid.Parse(entry.Name())
+		if err != nil {
+			continue
+		}
+
+		blockPath := filepath.Join(api.dbDir, entry.Name())
+		block, err := tsdb.OpenBlock(api.logger, blockPath, nil, nil)
```

Comment:
I assume opening every block in a large TSDB is still cheap enough to do all the time when looking at a UI page that displays it? As far as I can see, it opens and mmaps all the chunk files for each block as well (among other things), although we'd only need to read the block metadata. Would it make sense to instead expose and use only the `readMetaFile()` function from https://github.com/prometheus/prometheus/blob/4aee718013c52bef1ccb336bcc21886336e6252e/tsdb/block.go#L337C25-L337C37?